src/flick/lst/views.py

```diff
@@ -1,3 +1,160 @@
+            collaborator = User.objects.get(pk=c_id)
+            if Profile.objects.filter(user=collaborator):
+                c = Profile.objects.get(user=collaborator)
+                lst.collaborators.add(c)
+        for show_id in show_ids:
+            if Show.objects.filter(pk=show_id):
+                show = Show.objects.get(pk=show_id)
+                lst.shows.add(show)
+        lst.save()
+        serializer = LstSerializer(lst)
+        return success_response(serializer.data)
+
+
+class LstDetail(generics.GenericAPIView):
+
+    queryset = Lst.objects.all()
+    serializer_class = LstSerializer
+
+    permission_classes = api_settings.UNPROTECTED
+
+    def get(self, request, pk):
+        """Get a specific list by id."""
+        # TODO: only allow users who are owners / collaborators see lists
+        queryset = self.get_object()
+        serializer = LstSerializer(queryset, many=False)
+        return success_response(serializer.data)
+
+    def delete(self, request, pk):
+        """Delete a list by id."""
+        # TODO: only allow users who are owners delete a list
+        if not Lst.objects.filter(pk=pk):
+            return success_response(f"List of id {pk} has already been deleted.")
+        lst = Lst.objects.get(pk=pk)
+        lst.delete()
+        return success_response(f"List of id {pk} has been deleted.")
+
+    def post(self, request, pk):
+        """
+        Update a list by id.
+        Collaborators can update lst_pic, is_favorite, is_watched, collaborators, and shows.
+        An owner can update lst_name, lst_pic, is_favorite, is_private, is_watched, collaborators, the owner (can cede ownership completely to another user), and shows.
+        """
+        data = json.loads(request.body)
+        lst_name = data.get("lst_name")
+        lst_pic = data.get("lst_pic")
+        is_favorite = data.get("is_favorite", False)
+        is_private = data.get("is_private", False)
+        is_watched = data.get("is_watched", False)
+        collaborator_ids = data.get("collaborators", [])
+        owner_id = data.get("owner", request.user.id)
+        show_ids = data.get("shows", [])
+
+        if not Lst.objects.filter(pk=pk):
+            return failure_response(f"No list to be found with id of {pk}.")
+        lst = Lst.objects.get(pk=pk)
+
+        profile = Profile.objects.get(user=request.user)
+        if not profile:
+            return failure_response(f"No user to be found with id of {request.user.id}.")
+
+        user_is_collaborator = profile in lst.collaborators.all()
+        user_is_owner = profile == lst.owner
+
+        if not (user_is_collaborator or user_is_owner):
+            return failure_response(f"User {request.user} does not have the credentials to list of id {pk}.")
+        if user_is_collaborator:
+            lst.lst_pic = lst_pic
+            lst.is_favorite = is_favorite
+            lst.is_watched = is_watched
+            lst.collaborators.clear()
```

Comment:
are we not making list collaborators an invitation?

Reply:
ohhhh that's a super good point I didn't even think of that. hm, we can make that a new issue where we 1) detect who's been added / updated, 2) add invitation functionality #58 , but also curious to see what frontend ppl have to say about this